src/azul/service/async_manifest_service.py

```diff
@@ -63,40 +78,75 @@ class AsyncManifestService(AbstractService):
     """
     step_function_helper = StepFunctionHelper()
 
+    step_function_execution_retries = 2
+
     def __init__(self, state_machine_name):
         self.state_machine_name = state_machine_name
 
     def start_generation(self, input: JSON) -> Token:
-        execution_id = str(uuid.uuid4())
-        self.step_function_helper.start_execution(self.state_machine_name,
-                                                  execution_id,
-                                                  execution_input=input)
-        return Token(execution_id=execution_id,
-                     request_index=0,
-                     wait_time=self._get_next_wait_time(0))
+
+        def _start_generation(retry_index: int = 0) -> Token:
```

Comment:
I have the feeling the recursion can be converted to a loop over `range(step_function_execution_retries)`.